OrbitQt/orbitcodeeditor.cpp

```diff
@@ -438,6 +457,47 @@ void OrbitCodeEditor::lineNumberAreaPaintEvent(QPaintEvent* event) {
   }
 }
 
+//-----------------------------------------------------------------------------
+void OrbitCodeEditor::HeatMapAreaPaintEvent(QPaintEvent* event) {
+  QPainter painter(heatMapArea);
+  painter.fillRect(event->rect(), QColor(30, 30, 30));
+
+  //![extraAreaPaintEvent_0]
+
+  //![extraAreaPaintEvent_1]
+  QTextBlock block = firstVisibleBlock();
+  int blockNumber = block.blockNumber();
+  int top = static_cast<int>(
+      blockBoundingGeometry(block).translated(contentOffset()).top());
+  int bottom = top + static_cast<int>(blockBoundingRect(block).height());
+  //![extraAreaPaintEvent_1]
+
+  //![extraAreaPaintEvent_2]
+  while (block.isValid() && top <= event->rect().bottom()) {
+    if (block.isVisible() && bottom >= event->rect().top()) {
+      // % of hits in this function from 0.0 to 1.0
+      const double hit_ratio = line_to_hit_ratio_(blockNumber);
+      // Create a log-scale, values smaller or equal to 1% gets mapped to a
```

Comment:
Hmmm, this comment confused me a bit. 

You want a mapping that maps 0 -> 0 and that maps 1 to 1, but applies a log curve. So, all values get mapped to a value greater than 0. 

Suggestion: 
"Create a log scale such that a hit ratio of 0.0 is mapped to 0.0 and a hit ratio of 1.0 is mapped to 1.0. "

Reply:
Btw, a simple 

hit_ratio = sqrt(hit_ratio);

will give you something similar (might even be better to push up small sample numbers up a bit more).

Reply:
Thanks for the hint with the sqrt. This does its job and is more understandable. I adjusted the comment and made it to sqrt.